be/src/http/action/stream_load.cpp

```diff
@@ -168,6 +168,11 @@ Status StreamLoadAction::_handle(StreamLoadContext* ctx) {
         ctx->body_sink.reset();
         RETURN_IF_ERROR(_exec_env->stream_load_executor()->execute_plan_fragment(ctx));
     } else {
+        if (ctx->buf->pos > 0) {
+            ctx->buf->flip();
+            ctx->body_sink->append(std::move(ctx->buf));
+            ctx->buf = ByteBuffer::allocate(ctx->buf_default_sz);
```

Comment:
Q: When will the function `StreamLoadAction::_handle` be called? Is it still necessary to allocate a new buffer?

Reply:
Updated. The buffer is allocated on demand now.